db/skipped_sequence.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2016-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+const (
+	ClipCapacityHeadroom = 100
```

Comment:
Also, this value seems quite small to me, since we care about this most when the size of the slice is very large.  Thinking of a case where the length of the slice grows to 2.5M, then shrinks to 1M: the new capacity will be 1000100.  If only 101 sequences are added to the list, go is going to increase the capacity to 2000202.  I think we want a bit more buffer than that to minimize churn when the overall trend in the slice size is shrinking, but there's still some load.

Was there a specific reason for this value?

Reply:
Updated the headroom to be configurable. Also changed the default to be 1000. My thinking behind this was after looking at a few customer environments (that didn't have issues with skipped sequences) and the list never seemed to grown more than 30-40 in any environment I looked at. So thought I should have this capacity headroom fairly low. But you raise a good point, if we get a big jump in slice length this doesn't give much headroom for further jumps. 
Another option is to have two headroom values, one for when the list isn't getting used much < around 100 items say, and one for when slice grows > 10,000 for example.

Reply:
I agree that in most cases the list is small, but the memory associated with a minimum allocation of 1000 (or 10000) when the list is otherwise empty should be negligible, so I think a single value is fine.